spec/abstract/Tcb_A.thy

```diff
@@ -247,25 +247,25 @@ where
   (liftE $ do
     as_user tcb $ setRegister tlsBaseRegister tls_base;
     cur \<leftarrow> gets cur_thread;
-    when (tcb = cur) (do_extended_op reschedule_required);
+    when (tcb = cur) reschedule_required;
     return []
   od)"
 
 definition
-  set_domain :: "obj_ref \<Rightarrow> domain \<Rightarrow> unit det_ext_monad" where
+  set_domain :: "obj_ref \<Rightarrow> domain \<Rightarrow> (unit, 'z::state_ext) s_monad" where
   "set_domain tptr new_dom \<equiv> do
      cur \<leftarrow> gets cur_thread;
      tcb_sched_action tcb_sched_dequeue tptr;
      thread_set_domain tptr new_dom;
      ts \<leftarrow> get_thread_state tptr;
-     when (runnable ts) (tcb_sched_action tcb_sched_enqueue tptr);
-     when (tptr = cur) reschedule_required
+     when (runnable ts) $ tcb_sched_action tcb_sched_enqueue tptr;
+     when (tptr = cur) $ reschedule_required
```

Comment:
nitpick: this `$` is redundant

Reply:
Good spotting. This was copied from the rt branch, and I was originally thinking that it would be best to keep things the same even when the style isn't quite the best but I've since changed my mind and will clean up things like this. The merge back into rt is already going to be pretty bad, some extra clean up won't make it noticeably worse.